File: Java/Program/Product.java
/* Saya Syifa Azzahra NIM 2207308 mengerjakan soal Latihan 2 dalam mata kuliah 
Desain Pemrograman Berorientasi Objek untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang telah
dispesifikasikan. Aamiin. */

public class Product 
{
    private String produkID;
    private String nama;
    private String merk; 
    private String harga;

    // konstruktor default Product tanpa parameter
    public Product() 
    {
        
    }

    // konstruktor Product dengan parameter
    public Product(String produkID, String nama, String merk, String harga) 
    {
        this.produkID = produkID;
        this.nama = nama;
        this.merk = merk;
        this.harga = harga;
    }

    public String getProdukID() 
    {
        return produkID;
    }

    public void setProdukID(String produkID) 
    {
        this.produkID = produkID;
    }

    public String getNama() 
    {
        return nama;
    }

    public void setNama(String nama) 
    {
        this.nama = nama;
    }

    public String getMerk() 
    {
        return merk;
    }

    public void setMerk(String merk) 
    {
        this.merk = merk;
    }

    public String getHarga() 
    {
        return harga;
    }

    public void setHarga(String harga) 
    {
        this.harga = harga;
    }
}

File: CPP/Program/Product.cpp
/* Saya Syifa Azzahra NIM 2207308 mengerjakan soal Latihan 2 dalam mata kuliah 
Desain Pemrograman Berorientasi Objek untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang telah
dispesifikasikan. Aamiin. */

#include <string>

using namespace std; 

// mendefinisikan kelas Product
class Product 
{
    // atribut private
    private:
        string produkID;
        string nama;
        string merk;
        string harga;

    // atribut public
    public:
        // konstruktor default Product tanpa parameter
        Product() 
        {

        }

        // konstruktor Product dengan parameter
        Product(string produkID, string nama, string merk, string harga)
            : produkID(produkID), nama(nama), merk(merk), harga(harga) 
            {

            }

        // fungsi-fungsi untuk mengambil dan mengatur nilai atribut-atribut pada kelas Product
        string getProdukID() 
        {
            return produkID;
        }

        void setProdukID(string produkID) 
        {
            this->produkID = produkID;
        }

        string getNama() 
        {
            return nama;
        }

        void setNama(string nama) 
        {
            this->nama = nama;
        }

        string getMerk() 
        {
            return merk;
        }

        void setMerk(string merk) 
        {
            this->merk = merk;
        }

        string getHarga() 
        {
            return harga;
        }

        void setHarga(string harga) 
        {
            this->harga = harga;
        }
};